### ISSUE TITLE
Panic at 'Can't find MACRO_CALL@[..] in AstIdMap'

### ISSUE THREAD

**Issue by oxalica** [rust-lang/rust-analyzer#14534]
**rust-analyzer version**: Both 2023-03-27 and current master 01120f1213ad928de7300a8acf9f41bed72d0422
**rustc version**: 1.68.2

**relevant settings**: (eg. client settings, or environment variables like `CARGO`, `RUSTC`, `RUSTUP_HOME` or `CARGO_HOME`)

This panic is triggered in a very specific setup involving nested macros. Here's a simplified version to reproduce:
1. Have a trait with methods generated by a macro.
2. The (1) macro generates a method with a fully-qualified associated type of another type.
3. The type in (2) is from another macro.
4. Complete missing impls in the trait (1) impl causes a panic. The method is also missed from completion.
5. All of (1), (2) and (3) are necessary to reproduce.
```rust
macro_rules! ty { () => { i32 } }
trait SomeTrait { type Output; }
impl SomeTrait for i32 { type Output = i64; }
macro_rules! define_method {
    () => {
        fn method(&mut self, params: <ty!() as SomeTrait>::Output);
    };
}
trait AnotherTrait { define_method!(); }
impl AnotherTrait for () {
    // <- Trigger assist "Implement missing members" (add_missing_impl_members) to panic.
    // fn me| <- No completion.
}
```

May be relevant: #10388

<details>
<summary>Panic backtrace on 01120f1213ad928de7300a8acf9f41bed72d0422</summary>

```
thread 'handlers::add_missing_impl_members::tests::test_gen_trait_impl_with_macro_in_assoc_type' panicked at 'Can't find MACRO_CALL@29..34 in AstIdMap:
[SyntaxNodePtr { kind: MACRO_RULES, range: 0..33 }, SyntaxNodePtr { kind: TRAIT, range: 34..66 }, SyntaxNodePtr { kind: IMPL, range: 67..112 }, SyntaxNodePtr { kind: MACRO_RULES, range: 113..230 }, SyntaxNodePtr { kind: TRAIT, range: 231..271 }, SyntaxNodePtr { kind: IMPL, range: 272..422 }, SyntaxNodePtr { kind: TYPE_ALIAS, range: 52..64 }, SyntaxNodePtr { kind: TYPE_ALIAS, range: 92..110 }, SyntaxNodePtr { kind: MACRO_CALL, range: 252..269 }]', crates/hir-expand/src/ast_id_map.rs:131:21
stack backtrace:
   0: rust_begin_unwind
             at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/panicking.rs:575:5
   1: core::panicking::panic_fmt
             at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/core/src/panicking.rs:64:14
   2: hir_expand::ast_id_map::AstIdMap::erased_ast_id
   3: hir_expand::ast_id_map::AstIdMap::ast_id
   4: hir_def::body::LowerCtx::ast_id
   5: hir_def::hir::type_ref::TypeRef::from_ast
   6: hir_def::path::lower::lower_path
   7: hir_def::path::Path::from_src
   8: hir::semantics::SemanticsScope::speculative_resolve
   9: ide_db::path_transform::Ctx::transform_path
  10: ide_db::path_transform::Ctx::apply
  11: ide_db::path_transform::PathTransform::apply
  12: ide_assists::utils::add_trait_assoc_items_to_impl::{{closure}}
  13: core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &mut F>::call_once
  14: core::option::Option<T>::map
  15: <core::iter::adapters::map::Map<I,F> as core::iter::traits::iterator::Iterator>::next
  16: ide_assists::utils::add_trait_assoc_items_to_impl
  17: ide_assists::handlers::add_missing_impl_members::add_missing_impl_members_inner::{{closure}}
  18: ide_assists::assist_context::Assists::add::{{closure}}
  19: ide_assists::assist_context::Assists::add_impl
  20: ide_assists::assist_context::Assists::add
  21: ide_assists::handlers::add_missing_impl_members::add_missing_impl_members_inner
  22: ide_assists::handlers::add_missing_impl_members::add_missing_impl_members
  23: ide_assists::tests::check_with_config
  24: ide_assists::tests::check
  25: ide_assists::tests::check_assist
  26: ide_assists::handlers::add_missing_impl_members::tests::test_gen_trait_impl_with_macro_in_assoc_type
  27: ide_assists::handlers::add_missing_impl_members::tests::test_gen_trait_impl_with_macro_in_assoc_type::{{closure}}
  28: core::ops::function::FnOnce::call_once
  29: core::ops::function::FnOnce::call_once
             at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

</details>
